### PR TITLE
Support optional packageUri

### PR DESCRIPTION
**Pull request by jonasfj**
Ensure that we support cases where `packageUri` is not present.

Fixes https://github.com/dart-lang/tools/issues/146

### CODE REVIEW

File: pkgs/extension_discovery/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## 1.0.1-wip
 
 - Update the package description.
+- Support optional `packageUri`.
```
Comment (kenzieschmoll) [inline pkgs/extension_discovery/CHANGELOG.md:4]: if you merge master, 1.0.2-wip, is the latest changelog version. Maybe I shouldn't have added 1.0.2-wip in my commit? If not feel free to move the entry I added to 1.0.2-wip to 1.0.1-wip.

**Review by kenzieschmoll — APPROVED** [review]
lgtm with one comment

File: pkgs/extension_discovery/CHANGELOG.md (hunk 1 above)
Comment (jonasfj) [reply pkgs/extension_discovery/CHANGELOG.md:3]: `-wip` generally indicates that changes haven't been published.

So as is, we've just made changes to the package and not published any of them.

We can either manually roll into the Dart SDK repo (DEPS) file, or wait and see if Devon didn't do some awesome automation that does it magically.

Once in the Dart SDK repo, we can bump the version number in changelog and pubspec.yaml, and push a tag on github (or create a release on github) and trigger automated publishing.

Comment (jonasfj) [reply pkgs/extension_discovery/CHANGELOG.md:3]: I see that last update in DEPS was:
https://github.com/dart-lang/sdk/commit/f61107b4f6fb49e3ca329449b27002123dd7c017

**Pull request merged**